pandas/core/groupby.py

```diff
@@ -3674,10 +3674,25 @@ def transform(self, func, *args, **kwargs):
         Each subframe is endowed the attribute 'name' in case you need to know
         which group you are working on.
 
+        The current implementation imposes three requirements on f:
+
+        * f must return a value that either has the same shape as the input
+          subframe or can be broadcast to the shape of the input subframe.
+          For example, f returns a scalar it will be broadcast to have the
+          same shape as the input subframe.
+        * f must support application column-by-column in the subframe. If f 
+          also supports application to the entire subframe, then a fast path 
+          is used starting from the second chunk.
```

Comment:
can you add the same to SeriesGroupBy.transform?  or even better is to rip out the doc-strings to have a common one :> and use Appender.